### PR TITLE
[Removal] Don't remove calls to assert()

### PR DESCRIPTION
**Pull request by duncan3dc**
The new `FunctionCallRemoval` mutator is removing calls to `assert()` and expecting the tests to fail, but the `assert()` function is a special case that it doesn't make sense to remove.

This PR prevents calls to `assert()` from being removed, and adds a test to ensure they aren't removed

### CODE REVIEW

**Comment by duncan3dc** [comment]
I've made those changes now and the build is green :tada:

File: src/Mutator/Removal/FunctionCallRemoval.php [hunk 1]
```diff
@@ -28,6 +29,22 @@ protected function mutatesNode(Node $node): bool
...
 
-        return $node->expr instanceof Node\Expr\FuncCall;
+        if (!$node->expr instanceof Node\Expr\FuncCall) {
+            return false;
+        }
+
+        $name = $node->expr->name;
+
+        if (!$name instanceof Node\Name) {
+            return false;
```
Comment (BackEndTea) [inline src/Mutator/Removal/FunctionCallRemoval.php]: I think this can be `return true;`. Even if we don't know what function is called, because the name is a variable or a string for example, we can still remove it and see if it has an effect.

File: tests/Mutator/Removal/FunctionCallRemovalTest.php [hunk 2]
```diff
@@ -82,6 +82,17 @@ public function provideMutationCases(): \Generator
 
 $this->foo();
 $a = 3;
+PHP
+        ];
+
+        yield 'It does not remove an assert() call' => [
```
Comment (BackEndTea) [inline tests/Mutator/Removal/FunctionCallRemovalTest.php]: Please add a test where the function name is a variable or a string to make sure it doesn't crash there.

The `PregQuote` test is an example of how to do that.

File: src/Mutator/Removal/FunctionCallRemoval.php [hunk 3]
```diff
@@ -11,6 +11,7 @@
 
 use Infection\Mutator\Util\Mutator;
 use PhpParser\Node;
+use function strtolower;
```
Comment (BackEndTea) [inline src/Mutator/Removal/FunctionCallRemoval.php]: We don't do this anywhere within the code base, so please revert this change.

File: src/Mutator/Removal/FunctionCallRemoval.php (hunk 1 above)
Comment (duncan3dc) [reply src/Mutator/Removal/FunctionCallRemoval.php]: Yep, that was a mistake, fixed and added a test :+1:

File: src/Mutator/Removal/FunctionCallRemoval.php (hunk 3 above)
Comment (borNfreee) [reply src/Mutator/Removal/FunctionCallRemoval.php]: on the other side - we can discuss it and start using this approach, at least understand core developers think about it.

Personally, I find it more readable as it removes all those `\`,

`\assert(\count($a) > 0)` vs `assert(count($a) > 0)`

@infection/core

**Comment by borNfreee** [comment]
Nice work @duncan3dc ! Thank you for your contribution

**Comment by borNfreee** [comment]
Released https://github.com/infection/infection/releases/tag/0.10.1
